packages/api-gateway/src/services/purposeService.ts

```diff
@@ -61,11 +69,33 @@ const retrievePurpose = async (
       });
     });
 
+const retrieveActiveDelegationByEServiceId = async (
+  delegationProcessClient: DelegationProcessClient,
+  headers: ApiGatewayAppContext["headers"],
+  eserviceId: catalogApi.EService["id"]
+): Promise<delegationApi.Delegation | undefined> => {
+  const result = await delegationProcessClient.getDelegations({
+    headers,
+    queries: {
+      eserviceIds: [eserviceId],
+      kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
```

Comment:
Since the function does not include the name Producer I would pass this as a parameter

Reply:
Yes, I agree!  Thanks